### ISSUE TITLE
How to mirror the output of smart-contract to external DB?

### ISSUE THREAD

**Issue by manh-vv** [EOSIO/eosio-project-demux-example#27]
As I understand, flow for creating a post is:

1. Frontend side push a action `createpost` with compatible  payload to create new post. Then a corresponding smart-contract which was deployed on EOS will handle the action.

2. Backend side use `demux` pattern to listen on the same action `createpost` and get the payload then store the data to mongoDb.

That means the data we stored in blockchain may not be the same as data we stored on mongoDb. Because the procedures we apply to action payload is not the same. Is that right? 

If I am correct, do you have solution? Could you provide a sample?

**Comment by TaraTritt** [comment]
Yes, in this case of this app, we are only storing the author, auto-generated primary key, and composite key of the author and timestamp in the multi index table on the blockchain. In mongoDB we are storing all of our Dapp's state (blog - title, content, tag, author, timestamp). However, this is the intended behavior, by using demux and persisting state to a traditional database we can store the state of our Dapp in a structure that makes it easier to query and reduce the amount of data we need to store on the blockchain. If you wanted to store exactly what is in your multi index table in mongoDB, you would just need to replicate your multi index schema in mongoDB and implement your demux updater so that it updates mondoDB accordingly
